### PR TITLE
docs(common): update API docs for unified location service for upgrading

### PR DESCRIPTION
**Pull request by brandonroberts**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/common/upgrade/src/location_shim.ts [hunk 1]
```diff
@@ -514,16 +523,14 @@ export class $locationShim {
...
+   * A getter method that returns the path of the current URL when called without any parameter.
    *
-   * Return path of current URL when called without any parameter.
+   * When called with a defined paramter, it changes the path and returns a reference to its own
+   * instance.
    *
-   * Change path when called with parameter and return `$location`.
-   *
-   * Note: Path should always begin with forward slash (/), this method will add the forward slash
+   * Paths should always begin with forward slash (/). This method will add the forward slash
```
Comment (jbogarthyde) [inline packages/common/upgrade/src/location_shim.ts]: This method adds...

File: packages/common/upgrade/src/location_shim.ts [hunk 2]
```diff
@@ -548,11 +555,9 @@ export class $locationShim {
   }
 
   /**
-   * This method is getter / setter.
-   *
-   * Return search part (as object) of current URL when called without any parameter.
+   * A getter method that returns a map of the search parameters of the current URL when called without any parameter.
```
Comment (jbogarthyde) [inline packages/common/upgrade/src/location_shim.ts]: Retrieves a map of the search parameters of the current URL, or changes a search part and returns a reference to its own instance.

File: packages/common/upgrade/src/location_shim.ts [hunk 3]
```diff
@@ -585,8 +590,7 @@ export class $locationShim {
    * If `paramValue` is `true`, the property specified via the first argument will be added with no
    * value nor trailing equal sign.
    *
-   * @return {Object} If called with no arguments returns the parsed `search` object. If called with
-   * one or more arguments returns `$location` object itself.
+   * @return {Object} If no arguments are provided, it returns the parsed `search` object. Otherwise, it returns a reference to its own instance object itself.
```
Comment (jbogarthyde) [inline packages/common/upgrade/src/location_shim.ts]: The parsed `search` object of the current URL, or the changed `search` object.

File: packages/common/upgrade/src/location_shim.ts [hunk 4]
```diff
@@ -633,12 +637,11 @@ export class $locationShim {
   }
 
   /**
-   * This method is getter / setter.
-   *
-   * Returns the hash fragment when called without any parameters.
-   *
-   * Changes the hash fragment when called with a parameter and returns `$location`.
+   * A getter method that returns the hash fragment when called without any parameters.
```
Comment (jbogarthyde) [inline packages/common/upgrade/src/location_shim.ts]: Retrieves the current hash fragment, or changes hash fragment and returns the changed value.

**Review by jasonaden — APPROVED** [review]

**Comment by brandonroberts** [comment]
Caretaker note: This is a docs only change and should not impact g3. Please run presubmit.

**Pull request closed without merging**